src/qudi/hardware/adlink/fastcounter_adlink.py

```diff
@@ -0,0 +1,803 @@
+        self._measurement_buffer_address = ctypes.cast(
+            self._measurement_buffer, ctypes.c_void_p
+        )
+
+        err = AdlinkDataTypes.I16(
+            self._dll.WD_AI_CH_Config(
+                self._card,
+                AdlinkDataTypes.I16(-1),
+                AdlinkDataTypes.U16(self._settings.ad_range),
+            )
+        )
+        if self.check_if_error(err, "CH_Config"):
+            return
+
+        err = AdlinkDataTypes.I16(
+            self._dll.WD_AI_Config(
+                self._card,
+                self._settings.timebase,
+                self._settings.ad_duty_restore,
+                self._settings.ad_convert_source,
+                self._settings.double_edged,
+                self._settings.buf_auto_reset,
+            )
+        )
+        self.check_if_error(err, "Config")
+
+        err = AdlinkDataTypes.I16(
+            self._dll.WD_AI_Trig_Config(
+                self._card,
+                self._settings.ad_trigger_mode,
+                self._settings.ad_trigger_source,
+                self._settings.ad_trigger_polarity,
+                self._settings.analog_trigger_channel,
+                self._settings.analog_trigger_level,
+                self._settings.post_trigger_scans,
+                self._settings.pre_trigger_scans,
+                self._settings.trigger_delay_ticks,
+                AdlinkDataTypes.U32(0),
+            )
+        )
+        if self.check_if_error(err, "Trig_Config"):
+            return
+
+        # reserve memory using the DLL's function
+        self.buffer_size_bytes()
+        self._ai_buffer1 = ctypes.c_void_p(
+            self._dll.WD_Buffer_Alloc(self._card, self._buffer_size_bytes)
+        )
+
+        if self.check_if_error(
+            AdlinkDataTypes.I16(self._ai_buffer1.value), "BufferAlloc"
+        ):
+            return
+        ctypes.memset(self._ai_buffer1, 0, self._buffer_size_bytes.value)
+
+        err = AdlinkDataTypes.I16(
+            self._dll.WD_AI_ContBufferSetup(
+                self._card,
+                self._ai_buffer1,
+                self._buffer_size_samples,
+                ctypes.byref(self._buffer_id1),
+            )
+        )
+        if self.check_if_error(err, "ContBufferSetup"):
+            return
+
+        self._ai_buffer2 = ctypes.c_void_p(
+            self._dll.WD_Buffer_Alloc(self._card, self._buffer_size_bytes)
+        )
+
+        if self.check_if_error(
+            AdlinkDataTypes.I16(self._ai_buffer2.value), "BufferAlloc"
+        ):
+            return
+        ctypes.memset(self._ai_buffer2, 0, self._buffer_size_bytes.value)
+
+        err = AdlinkDataTypes.I16(
+            self._dll.WD_AI_ContBufferSetup(
+                self._card,
+                self._ai_buffer2,
+                self._buffer_size_samples,
+                ctypes.byref(self._buffer_id2),
+            )
+        )
+        if self.check_if_error(err, "ContBufferSetup"):
+            return
```

Comment:
This part can be simplified like:
  self._ai_buffer1 = ctypes.c_void_p(
            self._dll.WD_Buffer_Alloc(self._card, self._buffer_size_bytes)
        )
   self._configure_buffer(self._ai_buffer1, self._buffer_id1)

  self._ai_buffer2 = ctypes.c_void_p(
            self._dll.WD_Buffer_Alloc(self._card, self._buffer_size_bytes)
        )
   self._configure_buffer(self._ai_buffer2, self._buffer_id2)